test/integration/query/assert/allocation_allocationsummary_test.go

```diff
@@ -0,0 +1,108 @@
+package count
+
+// Description - Asserts Pod Results from /allocation match /allocation/summary
+
+import (
+	// "fmt"
+	// "time"
+	"testing"
+	"strings"
+	"github.com/opencost/opencost-integration-tests/pkg/api"
+)
+
+func assertPodInformation(apiSummary api.AllocationResponseItemProperties, apiCompute *api.AllocationResponseItemProperties) (bool) {
+	status := true
+	if apiSummary.Cluster != apiCompute.Cluster {
+		status = false
+	}
+	if apiSummary.Container != apiCompute.Container {
+		status = false
+	}
+	if apiSummary.Namespace != apiCompute.Namespace {
+		status = false
+	}
+	if apiSummary.Node != apiCompute.Node {
+		status = false
+	}
+	return status
+}
+func TestAllocationAndAllocationSummary(t *testing.T) {
+	apiObj := api.NewAPI()
+
+	testCases := []struct {
+		name		string
+		window      string
+		aggregate   string
+		accumulate  string
+	}{
+		{
+			name: "Yesterday",
+			window: "1d",
+			aggregate: "pod",
+			accumulate: "false",
+		},
+	}
+
+	t.Logf("testCases: %v", testCases)
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+
+			// API Client
+			apiResponse, err := apiObj.GetAllocation(api.AllocationRequest{
+				Window: tc.window,
+				Aggregate: tc.aggregate,
+			})
+
+			if err != nil {
+				t.Fatalf("Error while calling Allocation API %v", err)
+			}
+			if apiResponse.Code != 200 {
+				t.Errorf("API returned non-200 code")
+			}
+
+			apiSummaryResponse, err := apiObj.GetAllocationSummary(api.AllocationRequest{
+				Window: tc.window,
+				Aggregate: "pods", // Summary API is not working as expected
+			})
+
+			if err != nil {
+				t.Fatalf("Error while calling Allocation API %v", err)
+			}
+			if apiSummaryResponse.Code != 200 {
+				t.Errorf("API returned non-200 code")
+			}
+			
+			// Extract Key Information from Summary
+			var allPods map[string]api.AllocationResponseItemProperties = make(map[string]api.AllocationResponseItemProperties)
+			for key, _ := range apiSummaryResponse.Data.Sets[0].Allocations {
+				parts := strings.Split(key, "/")
+				if len(parts) == 5 {
+					cluster := parts[0]

```

Comment:
rather than attempting to tease out the different properties of the results and comparing them, just go in order through each API and compare the fields.

Reply:
allocation/summary does not return any unique identifier for me to check against /allocation. I am breaking down the name to get the pod name which is unique

Reply:
I guess my point is that you don't necessarily care about the pod name. 

Let's say that there are 3 namespaces, ns1, ns2,ns3. If you agg by namespace, you will get 3 results on each endpoint, one for ns1, ns2, ns3. those are what we want to compare - we are only interested in the costs of the namespace, not necessarily the pods in that namespace. if you did want to look at pods, you could agg by pod and filter by namespace?